common/persistence/persistenceMetricClients.go

```diff
@@ -224,54 +224,99 @@ func NewConfigStorePersistenceMetricsClient(
+		metricsScope.IncCounter(metrics.PersistenceErrConditionFailedCounter)
 	case *CurrentWorkflowConditionFailedError:
-		scopeWithDomainTag.IncCounter(metrics.PersistenceErrCurrentWorkflowConditionFailedCounter)
+		metricsScope.IncCounter(metrics.PersistenceErrCurrentWorkflowConditionFailedCounter)
 	case *ShardAlreadyExistError:
-		scopeWithDomainTag.IncCounter(metrics.PersistenceErrShardExistsCounter)
+		metricsScope.IncCounter(metrics.PersistenceErrShardExistsCounter)
 	case *ShardOwnershipLostError:
-		scopeWithDomainTag.IncCounter(metrics.PersistenceErrShardOwnershipLostCounter)
+		metricsScope.IncCounter(metrics.PersistenceErrShardOwnershipLostCounter)
 	case *types.EntityNotExistsError:
-		scopeWithDomainTag.IncCounter(metrics.PersistenceErrEntityNotExistsCounter)
+		metricsScope.IncCounter(metrics.PersistenceErrEntityNotExistsCounter)
 	case *TimeoutError:
-		scopeWithDomainTag.IncCounter(metrics.PersistenceErrTimeoutCounter)
-		scopeWithDomainTag.IncCounter(metrics.PersistenceFailures)
+		metricsScope.IncCounter(metrics.PersistenceErrTimeoutCounter)
+		metricsScope.IncCounter(metrics.PersistenceFailures)
 	case *types.ServiceBusyError:
-		scopeWithDomainTag.IncCounter(metrics.PersistenceErrBusyCounter)
-		scopeWithDomainTag.IncCounter(metrics.PersistenceFailures)
+		metricsScope.IncCounter(metrics.PersistenceErrBusyCounter)
+		metricsScope.IncCounter(metrics.PersistenceFailures)
 	case *DBUnavailableError:
-		scopeWithDomainTag.IncCounter(metrics.PersistenceErrDBUnavailableCounter)
-		scopeWithDomainTag.IncCounter(metrics.PersistenceFailures)
+		metricsScope.IncCounter(metrics.PersistenceErrDBUnavailableCounter)
+		metricsScope.IncCounter(metrics.PersistenceFailures)
 		p.logger.Error("DBUnavailable Error:", tag.Error(err), tag.MetricScope(scope))
 	default:
 		p.logger.Error("Operation failed with internal error.", tag.Error(err), tag.MetricScope(scope))
-		scopeWithDomainTag.IncCounter(metrics.PersistenceFailures)
+		metricsScope.IncCounter(metrics.PersistenceFailures)
 	}
 }
 
 func (p *persistenceMetricsClientBase) call(scope int, op func() error, tags ...metrics.Tag) error {
-	scopeWithDomainTag := p.metricClient.Scope(scope, tags...)
-	scopeWithDomainTag.IncCounter(metrics.PersistenceRequests)
+	metricsScope := p.metricClient.Scope(scope, tags...)
+	if len(tags) > 0 {
+		metricsScope.IncCounter(metrics.PersistenceRequestsPerDomain)
+	} else {
+		metricsScope.IncCounter(metrics.PersistenceRequests)
+	}
 	before := time.Now()
 	err := op()
 	duration := time.Since(before)
-	scopeWithDomainTag.RecordTimer(metrics.PersistenceLatency, duration)
+	if len(tags) > 0 {
+		metricsScope.RecordTimer(metrics.PersistenceLatencyPerDomain, duration)
+	} else {
+		metricsScope.RecordTimer(metrics.PersistenceLatency, duration)
+	}
 
 	if p.enableLatencyHistogramMetrics {
-		scopeWithDomainTag.RecordHistogramDuration(metrics.PersistenceLatencyHistogram, duration)
+		metricsScope.RecordHistogramDuration(metrics.PersistenceLatencyHistogram, duration)
```

Comment:
I think it's not necessary to introduce per domain metrics for histogram.

Reply:
if anything I think histograms are candidates for emitting the new tag + not creating a new name.  histograms aggregate correctly, timers don't.